src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2687,6 +2687,10 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                         Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, distinguisher.First, distinguisher.Second);
                     }
                 }
+                else if (sourceType.SpecialType == SpecialType.System_Void)
+                {
+                    Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, sourceType.ToDisplayString(), targetType.ToDisplayString());
```

Comment:
I believe `ToDisplayString()` is the default formatting for `Symbol`, and the other instance of this `ErrorCode` above uses `SymbolDistinguisher` which generates strings lazily. I agree with @AlekseyTs, we should pass `sourceType` and `targetType` as arguments and generate strings only when the message is formatted.